src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy strategy;
+    private HashMap storage;
+
+    public ShopServiceImpl() {
+        strategy = new OperationStrategyImpl();
+        storage = new HashMap<String, Integer>();
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            try {
+                strategy.chooseHandler(transaction).handle(storage, transaction);
+            } catch (RuntimeException e) {
+                throw new RuntimeException("Error processing transaction: " + e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getStorage() {
+        return (Map<String, Integer>) storage.clone();
+    }
```

Comment:
Returning a cloned copy of the storage map can be misleading and potentially dangerous if the clone is modified. Consider returning an unmodifiable view of the map using `Collections.unmodifiableMap(storage)` to ensure the internal state of `ShopServiceImpl` cannot be altered.